geek.txt/backend/routes/users.js

```diff
@@ -1,59 +1,99 @@
-// 
 /*
    API routes related to the User model.
-   Document the routes here...
-
-   (get) /
-        body: {
-
-        }
-
-   (post) /add
-        body: {
-            "username":
-            "password":
-        }
-    ...
-    
 */
 
 const router = require('express').Router();
-let User = require('../models/users.model')
+const bcrypt = require('bcrypt');
+let Users = require('../models/users.model');
+let EmailValidator = require('./utils/validators')
 
-// Route to get all users.
-
-// Handles incomming GET requests to url/users/ .
+// Generate Salt
+const salt = bcrypt.genSaltSync(10);
+    
+// Handles incomming GET requests to users/.
 router.route('/').get((req, res) => {
     // List of all users in the DB.
-    User.find()
+    Users.find()
         // If promise return then return all users as JSON.
         .then(users => res.json(users))
         // If there is an error return status 400 with Error.
         .catch(err => res.status(400).json('Error: ' + err))
-})
+});
 
-// Handle post request to add new user.
-// Handles incomming post request to url/users/add 
-router.route('/add').post((req, res) => {
-    /* 
-    Sample request body
-    {
-        "username": "SampleUser",
-        "password": "SamplePass"
-    }
+
+// Handles incomming POST requests to users/login.
+router.route('/login').post((req, res) => {
+    /*
+    Sample POST request:
+        "credential" : "testUser",
+        "password" : "User"
     */
+    
+    // Can be an email or a geek ID.
+    const credential = req.body.credential
+    const password = req.body.password
+    
+    console.log(password)
+
+    // Hash Password
+    const hashed_password = bcrypt.hashSync(password, salt);
 
-    const username = req.body.username;
-    const password = req.body.password;
+    console.log(hashed_password)
+    if(EmailValidator.validateEmail(credential)){
+        // Lookup user by email.
+        Users.find({email: credential, password: hashed_password})
+        // If promise return then return all users as JSON.
+        .then(user => res.json(user))
```

Comment:
Here is where you would need to check if the password matches that hashed password in the database. If not return an error saying invalid or something.